exchanges/gateio/gateio_ws_futures.go

```diff
@@ -514,139 +511,120 @@ func (g *Gateio) processFuturesCandlesticks(data []byte, assetType asset.Item) e
 	return nil
 }
 
-func (g *Gateio) processFuturesOrderbookTicker(data []byte) error {
-	var response WsResponse
-	orderbookTicker := &WsFuturesOrderbookTicker{}
-	response.Result = orderbookTicker
-	err := json.Unmarshal(data, &response)
+func (g *Gateio) processFuturesOrderbookTicker(incoming []byte) error {
+	var data WsFuturesOrderbookTicker
+	err := json.Unmarshal(incoming, &data)
 	if err != nil {
 		return err
 	}
-	g.Websocket.DataHandler <- response
+	g.Websocket.DataHandler <- data
 	return nil
 }
 
-func (g *Gateio) processFuturesAndOptionsOrderbookUpdate(data []byte, assetType asset.Item) error {
-	var response WsResponse
-	update := &WsFuturesAndOptionsOrderbookUpdate{}
-	response.Result = update
-	err := json.Unmarshal(data, &response)
+func (g *Gateio) processFuturesAndOptionsOrderbookUpdate(incoming []byte, assetType asset.Item) error {
+	var data WsFuturesAndOptionsOrderbookUpdate
+	err := json.Unmarshal(incoming, &data)
 	if err != nil {
 		return err
 	}
-	pair, err := currency.NewPairFromString(update.ContractName)
+	pair, err := currency.NewPairFromString(data.ContractName)
```

Comment:
I'm not sure exactly whats causing it, but actions tests are showing 
` gateio_test.go:2900: GateIO websocket options orderbook update push data error: cannot create currency pair from  string too short to be a currency pair`
and `data` is an empty struct at this point

Reply:
🦅